autogen/agentchat/contrib/society_of_mind_agent.py

```diff
@@ -0,0 +1,176 @@
+                If response_preparer is a string, then it should be the LLM prompt used to extract the final
+                message from the inner chat transcript.
+                The default response_preparer depends on if an llm_config is provided. If llm_config is False,
+                then the response_preparer deterministically returns the last message in the inner-monolgue. If
+                llm_config is set to anything else, then a default LLM prompt is used.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        chat_manager: GroupChatManager,
+        response_preparer: Optional[Union[str, Callable]] = None,
+        is_termination_msg: Optional[Callable[[Dict], bool]] = None,
+        max_consecutive_auto_reply: Optional[int] = None,
+        human_input_mode: Optional[str] = "TERMINATE",
+        function_map: Optional[Dict[str, Callable]] = None,
+        code_execution_config: Optional[Union[Dict, Literal[False]]] = None,
+        llm_config: Optional[Union[Dict, Literal[False]]] = False,
+        default_auto_reply: Optional[Union[str, Dict, None]] = "",
+    ):
+        super().__init__(
+            name=name,
+            system_message="",
+            is_termination_msg=is_termination_msg,
+            max_consecutive_auto_reply=max_consecutive_auto_reply,
+            human_input_mode=human_input_mode,
+            function_map=function_map,
+            code_execution_config=code_execution_config,
+            llm_config=llm_config,
+            default_auto_reply=default_auto_reply,
+        )
+
+        self.update_chat_manager(chat_manager)
+
+        # response_preparer default depends on if the llm_config is set, and if a client was created
+        if response_preparer is None:
+            if self.client is not None:
+                response_preparer = "Output a standalone response to the original request, without mentioning any of the intermediate discussion."
+            else:
+
+                def response_preparer(agent, messages):
+                    return messages[-1]["content"].replace("TERMINATE", "").strip()
+
+        # Create the response_preparer callable, if given only a prompt string
+        if isinstance(response_preparer, str):
+            self.response_preparer = lambda agent, messages: agent._lmm_response_preparer(response_preparer, messages)
+        else:
+            self.response_preparer = response_preparer
+
+        # NOTE: Async reply functions are not yet supported with this contrib agent
+        self._reply_func_list = []
+        self.register_reply([Agent, None], SocietyOfMindAgent.generate_inner_monologue_reply)
+        self.register_reply([Agent, None], ConversableAgent.generate_code_execution_reply)
+        self.register_reply([Agent, None], ConversableAgent.generate_function_call_reply)
+        self.register_reply([Agent, None], ConversableAgent.check_termination_and_human_reply)
+
+    def _lmm_response_preparer(self, prompt, messages):
```

Comment:
`_lmm_response_preparer` => "_llm_response_preparer"? as pointed by Beibin in the other PR.

Reply:
haha yeah